### PR TITLE
Add CI GitHub Action

### PR DESCRIPTION
**Pull request by MikeRogers0**
This is the GitHub action I used on https://github.com/MikeRogers0/faraday-net_http_persistent/blob/master/.github/workflows/ci.yml , which I originally copied from https://github.com/lostisland/faraday-net_http/blob/main/.github/workflows/ci.yml :)

The main change I made was to use Standard over Rubocop for linting as I find it more forgiving.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      - name: Standard
+        run: |
+          gem install standardrb
+          bundle exec standardrb
+  build:
+    needs: [linting]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        ruby: ['2.5', '2.6', '2.7']
```
Comment (olleolleolle) [inline .github/workflows/ci.yml]:
```suggestion
        ruby: ['2.5', '2.6', '2.7', '3.0']
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches: [master]
```
Comment (olleolleolle) [inline .github/workflows/ci.yml]: Perhaps we can make these run on all branches and all PRs? That way, we don't assume people's name of the default branch.

**Review by olleolleolle — COMMENTED** [review]
Thanks, @MikeRogers0 for starting this!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (MikeRogers0) [reply .github/workflows/ci.yml]: @olleolleolle Updated :) I also included `head` so it always tests the latest versions.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (MikeRogers0) [reply .github/workflows/ci.yml]: Agreed - I made it so it'll run whenever someone pushes :)

**Comment by MikeRogers0** [comment]
@olleolleolle Updated :)

FYI https://github.com/MikeRogers0/faraday-adapter-template/actions/runs/707904772 Here is a preview of it running. Right now the default specs do fail, I'm guessing because it's just a sample on how to do them :)

**Review by Heheehd — APPROVED** [review]

**Pull request merged**